### PR TITLE
Bad Metadata transferred for deserialisation [ Mintable(ERC721) ]

### PR DESCRIPTION
**Pull request by itzmeanjan**
## Changes

Before passing metadata to L1 token ( from L1 predicate ), deserialise it properly ( into `bytes` )

@rahuldamodar94 helped in catching this issue

### CODE REVIEW

File: test/predicates/MintableERC721Predicate.test.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ import * as deployer from '../helpers/deployer'
 import { mockValues } from '../helpers/constants'
 import logDecoder from '../helpers/log-decoder.js'
 import { getERC721TransferLog, getERC721TransferWithMetadataLog } from '../helpers/logs'
+import { it } from 'ethers/wordlists'
```
Comment (imyourm8) [inline test/predicates/MintableERC721Predicate.test.js:12]: this is going to break tests, remove it please

**Review by imyourm8 — CHANGES_REQUESTED** [review]
I think a test with metadata is missing.

File: test/predicates/MintableERC721Predicate.test.js (hunk 1 above)
Comment (itzmeanjan) [reply test/predicates/MintableERC721Predicate.test.js:12]: Yeah, done

**Comment by itzmeanjan** [comment]
> I think a test with metadata is missing.

You mean for ERC721Predicate ? I'll write it.

**Comment by itzmeanjan** [comment]
Updated test cases.

**Review by imyourm8 — APPROVED** [review]

**Pull request merged**